### PR TITLE
Allow DockerHub username to be specified via `DOCKER_USERNAME` var

### PR DESCRIPTION
**Pull request by kevin-david**
This was useful because I was impatient in waiting for #95 to be merged :) 

Passing CI: https://github.com/kevin-david/neolink/actions/runs/5100759096

### CODE REVIEW

File: .github/workflows/dockerprune.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ jobs:
             if [ -n $SECRET ]; then HAS_SECRET='true' ; fi
             echo "HAS_SECRET_TOKEN=${HAS_SECRET}" >> $GITHUB_OUTPUT
         env:
-          SECRET: "${{ secrets.DOCKER_PRUNE_TOKEN }}"
```
Comment (QuantumEntangledAndy) [inline .github/workflows/dockerprune.yml:34]: Please don't change this I have separate token scopes for it.

File: .github/workflows/dockerprune.yml [hunk 2]
```diff
@@ -143,5 +143,5 @@ jobs:
         env:
           DAYS: "100"
           REPO_NAME: ${{ steps.string_repo.outputs.lowercase }}
-          OWNER: ${{ steps.string_user.outputs.lowercase }}
```
Comment (QuantumEntangledAndy) [inline .github/workflows/dockerprune.yml:146]: Probably should just leave the prune script alone it's not needed for building or anything. It's used for cleaning up old images from dockerhub

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -195,8 +195,23 @@ jobs:
...
+          if [ -z $DOCKER_USERNAME ]; then
+            echo "DOCKER_NWO=${GITHUB_NWO}" >> $GITHUB_OUTPUT
+          else
+            IFS='/' read -ra NWO_SPLIT <<< "$GITHUB_NWO"
+            REPO_NAME=${NWO_SPLIT[1]}
+            echo "DOCKER_NWO=${DOCKER_USERNAME}/${REPO_NAME}" >> $GITHUB_OUTPUT
+          fi
+        env:
+          DOCKER_USERNAME: ${{ vars.DOCKER_USERNAME }}
+          GITHUB_NWO: ${{ steps.string_repo.outputs.lowercase }}
```
Comment (QuantumEntangledAndy) [inline .github/workflows/build.yml:214]: Why split it here. Why not use `string_user`

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -272,7 +287,7 @@ jobs:
           push: true
           file: Dockerfile
           tags: ${{ steps.tags.outputs.TAGS }}
-          build-args: VERSION=${{ steps.toml.outputs.version }}, REPO=${{ github.repository }}, OWNER=${{ github.repository_owner }}
+          build-args: VERSION=${{ steps.toml.outputs.version }}, REPO=${{ github.repository }}, OWNER=${{ vars.DOCKER_USERNAME || github.repository_owner }}
```
Comment (QuantumEntangledAndy) [inline .github/workflows/build.yml:290]: This || is a Boolean comparison. Are you sure it will work this way?

Comment (kevin-david) [reply .github/workflows/build.yml:290]: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-a-fallback-value-1

File: .github/workflows/dockerprune.yml (hunk 1 above)
Comment (kevin-david) [reply .github/workflows/dockerprune.yml:34]: also - any thoughts on _why_ have it's worth having token with different scopes? they're both repo-level secrets presumably, unless you're using them for something else? maybe i should be doing the same...

Comment (QuantumEntangledAndy) [reply .github/workflows/dockerprune.yml:34]: Might have a couple typos in should check. I just like seperate tokens for different things. Also the DOCKER_TOKEN I generated is super old and dosent seem to have the right privileges to do what I want in that script. Would always throw a access denied until I generated that new token just for pruning.

File: .github/workflows/build.yml (hunk 3 above)
Comment (QuantumEntangledAndy) [reply .github/workflows/build.yml:214]: Isn't it `github.repository_owner` I'm using it above (after conversion to lowercase coz docker hub wanted lower case when I registered) for my username.

Comment (QuantumEntangledAndy) [reply .github/workflows/build.yml:214]: Oh wait your after the repo name?

Comment (QuantumEntangledAndy) [reply .github/workflows/build.yml:214]: I think that's what I get for checking code while travelling

Comment (kevin-david) [reply .github/workflows/build.yml:214]: right - basically just the repo name to combine with the docker username

File: .github/workflows/dockerprune.yml (hunk 1 above)
Comment (kevin-david) [reply .github/workflows/dockerprune.yml]: Switched back to `DOCKER_PRUNE_TOKEN` and fixed the typo

**Comment by QuantumEntangledAndy** [comment]
Sorry for the delay on holiday at the moment so not much time for this sort of thing. I'll merge this one now

**Comment by kevin-david** [comment]
Thanks, no problem at all - I'm obviously in no rush given how long my first PR took! Hope you enjoyed the time off.

**Pull request merged**